### PR TITLE
[BUGFIX] OnboardingDataAssistant and Underlying Metrics: Add Defensive Programming Into Metric Implementations So As To Avoid Warnings About Incompatible Data

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Fix `ColumnValuesLengthMin` and `ColumnValuesLengthMax` metric providers to filter out `NULL` columns (cannot compute length of a string that turns into a floating point `NaN` value).
* Fix `ColumnMedian` metric provider to exit uniformly when all values in the column are `NULL` (return `0.0` in this case).
* Update `CANDIDATE_REGEX` in `RegexPatternStringParameterBuilder` to use Matching Groups instead of Capture Groups (the latter generate an "advice" warning from `Python` `RegEx` library).

### Remark
This fixes most of the warnings.  Next is to address warnings and errors, associated with `histogram` computation as well as any remaining exceptions in `Chart` graphing.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-822/GREAT-969
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py [hunk 1]
```diff
@@ -27,6 +27,11 @@ class ColumnMedian(ColumnAggregateMetricProvider):
     @column_aggregate_value(engine=PandasExecutionEngine)
     def _pandas(cls, column, **kwargs):
         """Pandas Median Implementation"""
+        metrics: Dict[str, Any] = kwargs.get("metrics") or {}
```
Comment (cdkini) [inline great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py:30]: Could we use `get("metrics", {})` instead?

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py [hunk 2]
```diff
@@ -27,6 +27,11 @@ class ColumnMedian(ColumnAggregateMetricProvider):
...
+        nonnull_count = metrics.get("column_values.nonnull.count")
+        if not nonnull_count:
+            return 0.0
```
Comment (cdkini) [inline great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py:33]: Nitpick - more personal preference but when not actually using the value, I think `if x not in metrics` is more appropriate. Happy to leave it as it though.

**Review by cdkini — APPROVED** [review]

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py:30]: @cdkini I believe that we could do that.  I will update in the next PR I am working on.  Thank you!

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/column_aggregate_metrics/column_median.py:33]: @cdkini I believe that here, even if it is there but returns `0` (or `None`), the check must return `True` (so this pattern, which we have everywhere for `metrics`, covers all of these needs).  Thanks!

**Pull request merged**